packages/svelte/src/internal/client/proxy.js

```diff
@@ -51,11 +55,23 @@ export function proxy(value, immutable = true) {
 					/** @type {import('./types.js').ProxyStateObject<T>} */ (proxy),
 					immutable
 				),
-				writable: false
+				writable: true,
+				enumerable: false
 			});
 
 			return proxy;
 		}
+	} else if (
+		DEV &&
+		typeof value === 'object' &&
+		value != null &&
+		is_frozen(value) &&
+		STATE_SYMBOL in value
+	) {
+		// eslint-disable-next-line no-console
+		console.warn(
+			'Object passed to $state was frozen after being passed to $state before. Consider using $state.frozen instead, or unfreeze the object.'
```

Comment:
I find this quite hard to parse, and IIUC it's not entirely accurate since in the accompanying test `$state` is being called with a brand new object that happens to have some extra symbols. As far as I can tell, we get the right outcome anyway — what exactly are we warning against here?

Reply:
Yeah the warning is tough to read, open for better suggestions. My fear is that there are cases where the freeze will prevent proper checks/updates of the state object: if it's stale you can't reset it because the object is frozen. I'll try to come up with an example tomorrow (maybe I'm wrong and it's fine)

Reply:
Removed it, couldn't find a way to break it